src/julia_internal.h

```diff
@@ -291,7 +291,7 @@ jl_value_t *jl_permbox32(jl_datatype_t *t, int32_t x);
 jl_value_t *jl_permbox64(jl_datatype_t *t, int64_t x);
 jl_svec_t *jl_perm_symsvec(size_t n, ...);
 
-#if !defined(__clang_analyzer__) // this sizeof(__VA_ARGS__) trick can't be computed until C11, but only the analyzer seems to care
+#if !defined(_COMPILER_CLANG_) // this sizeof(__VA_ARGS__) trick can't be computed until C11, but only Clang seems to care (when analyzing or using ASAN)
```

Comment:
I would prefer to keep this protection if at all possible. Not sure why ASAN breaks it but if it works fine with regular clang, can we just disable it under asan? Alternatively, we could set our default C standard to C11.